build.xml

```diff
@@ -132,17 +132,76 @@
 	</target>
 
     <target name="prodVersion">
-            <!-- Full version examples: 1.0.0-SNAPSHOT-0-HASH, 1.0.0-0-HASH, 1.0.0-1-HASH, etc -->
-            <exec outputproperty="fullVersion" dir="." executable="sh">
-			<arg value="-c" />
-                <arg value="git describe --long | tr -d v" />
-		</exec>
-            <!-- Version examples: 1.0.0-SNAPSHOT, 1.0.0, etc -->
-            <exec outputproperty="releaseVersion" dir="." executable="sh">
-                <arg value="-c" />
-                <arg value="git describe --abbrev=0 | tr -d v" />
-		</exec>
-            <echo>Full Version: ${fullVersion}</echo>
-            <echo>Version: ${releaseVersion}</echo>
-	</target>
+        <!-- Full version examples: 1.0.0-SNAPSHOT-0-HASH, 1.0.0-0-HASH, 1.0.0-1-HASH, etc -->
+        <exec outputproperty="fullVersion" dir="." executable="sh">
+            <arg value="-c" />
+            <arg value="git describe --long | tr -d v" />
+        </exec>
+        <!-- Version examples: 1.0.0-SNAPSHOT, 1.0.0, etc -->
+        <exec outputproperty="releaseVersion" dir="." executable="sh">
+            <arg value="-c" />
+            <arg value="git describe --abbrev=0 | tr -d v" />
+        </exec>
+        <echo>Full Version: ${fullVersion}</echo>
+        <echo>Version: ${releaseVersion}</echo>
+    </target>
+
+    <target name="dockerImageBuild" depends="prodVersion,createBuildDir,copyAndUnzipServerUpgradeBundle,copyServerUpgradeBundle,createDockerImage" description="Copy result files to build dir">
+        <zip destfile="Build${fullVersion}/OSC-${releaseVersion}_Build${fullVersion}.zip">
+            <fileset dir="Build${fullVersion}" includes="**/*.tar **/serverUpgradeBundle-*.zip" />
+        </zip>
+    </target>
+
+    <target name="checkServerUpgradeBundleExist">
+        <available file="docker/serverUpgradeBundle*.zip" property="isServerUpgradeBundleExist"/>
+    </target>
+
+    <target name="deleteOldServerUpgradeBundle" depends="checkServerUpgradeBundleExist" if="isServerUpgradeBundleExist">
+        <delete>
+            <fileset dir="docker/" includes="serverUpgradeBundle-*.zip" />
+        </delete>
+    </target>
+
+    <target name="copyAndUnzipServerUpgradeBundle" depends="deleteOldServerUpgradeBundle">
+        <copy file="target/serverUpgradeBundle.zip" tofile="docker/serverUpgradeBundle-${fullVersion}.zip" />
+            <exec executable="unzip" resultproperty="tar.result" dir="docker/">
+                <arg value="serverUpgradeBundle-${fullVersion}.zip" />
+            </exec>
+    </target>
+
+    <target name="clean" >
+        <delete file="docker/serverUpgradeBundle-${fullVersion}.zip"/>
+        <delete dir="docker/opt" />
+    </target>
+
+    <target name="createDockerImage" description="create docker image">
+        <exec executable="docker" resultproperty="tar.result">
+            <arg value="pull" />
+            <arg value="centos:7" />
```

Comment:
why is the version specified? I presume we use latest available centos from centos repository.

Reply:
Done, now docker pulls latest centos image from docker repository